src/odemis/gui/cont/tabs.py

```diff
@@ -4455,7 +4469,497 @@ def _on_align_move_done(self, f=None):
+                                             lambda evt: self._sem_alignment_controls("x", -1, evt),
+                                         panel.beam_shift_btn_p_aligner_x:
+                                             lambda evt: self._sem_alignment_controls("x", 1, evt),
+                                         panel.beam_shift_btn_m_aligner_y:
+                                             lambda evt: self._sem_alignment_controls("y", -1, evt),
+                                         panel.beam_shift_btn_p_aligner_y:
+                                             lambda evt: self._sem_alignment_controls("y", 1, evt),
+                                         }
+        flm_aligner_control_functions = {panel.flm_align_btn_m_aligner_x:
+                                             lambda evt: self._flm_alignment_controls("x", -1, evt),
+                                         panel.flm_align_btn_p_aligner_x:
+                                             lambda evt: self._flm_alignment_controls("x", 1, evt),
+                                         panel.flm_align_btn_m_aligner_y:
+                                             lambda evt: self._flm_alignment_controls("y", -1, evt),
+                                         panel.flm_align_btn_p_aligner_y:
+                                             lambda evt: self._flm_alignment_controls("y", 1, evt),
+                                         panel.flm_align_btn_m_aligner_z:
+                                             lambda evt: self._flm_alignment_controls("z", -1, evt),
+                                         panel.flm_align_btn_p_aligner_z:
+                                             lambda evt: self._flm_alignment_controls("z", 1, evt),
+                                         }
+
+        self._combined_aligner_control_functions = {**z_aligner_control_functions,
+                                                    **sem_aligner_control_functions,
+                                                    **flm_aligner_control_functions}
+
+        # Bind alignment control buttons to defined control functions
+        for btn, function in self._combined_aligner_control_functions.items():
+            btn.Bind(wx.EVT_BUTTON, function)
+
+        # Vigilant attribute connectors for the slider
+        self._step_size_controls_va_connector = VigilantAttributeConnector(tab_data.step_size,
+                                                                           self.panel.controls_step_size_slider,
+                                                                           events=wx.EVT_SCROLL_CHANGED)
+
+
+        panel.btn_custom_alignment.Bind(wx.EVT_BUTTON, self._on_click_custom_alignment)
+
+        # Bind the align mode buttons
+        for btn in self._align_btn_functions:
+            btn.Bind(wx.EVT_BUTTON, self._click_align_mode_button)
+
+        # Start with the first mode as default
+        self._align_btn_functions[panel.btn_align_z]()
```

Comment:
I'd suggest to handle this a little bit differently:
* When one of the alignment button is pressed, set .align_mode to the corresponding mode (you can do this in just a couple of lines with a dict).
* Have a subscriber to the .align_mode (you can do that here, and pass `init=True`). Then, this function does 2 things: update the toggle buttons to the right state ( as currently done in `_click_align_mode_button()`), and set the tab to the proper state, essentially by calling the corresponding `_set_XXX_alignment_mode()` function. You can dispatch either with a if/elif/else block, or a dict. As there are only 3 functions, I feel that an if/elif block is fine.